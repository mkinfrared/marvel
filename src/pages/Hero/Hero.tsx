import React from "react";
import { useParams } from "react-router-dom";

import HeroDetail from "containers/HeroDetail";

type Params = {
  heroId: string;
};

const Hero = () => {
  const { heroId } = useParams<Params>();
  const id = parseInt(heroId, 0);

  return <HeroDetail id={id} />;
};

export default React.memo(Hero);
