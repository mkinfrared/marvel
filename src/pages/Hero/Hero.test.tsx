import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import Hero from "./Hero";

jest.mock("containers/HeroDetail", () => () => "<HeroDetail />");

describe("<Hero />", () => {
  const Component = (
    <MemoryRouter>
      <Hero />
    </MemoryRouter>
  );

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
