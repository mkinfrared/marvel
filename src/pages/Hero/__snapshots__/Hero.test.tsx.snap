// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> match the snapshot 1`] = `
<div>
  &lt;HeroDetail /&gt;
</div>
`;
